Instrument event for order completion

sikachu opened the pull request:
This event will be instrumented when order has been confirmed by the customer during the checkout process.

elle commented:
Since you are checking for `payload` in one spec, does it make sense to check for `payload` in the other spec as well?

Also, funny that their state machine uses `.next!` rather than `.complete` but just semantics.

Other than that one comment, LGTM

sikachu commented:
@elle Yep, fixed that. I think they are more similar now.

Also, I fixed the random failure due to the fact that the subscribe block on the first example actually got triggered when the second example ran. That was an interesting parallel computing problem ... ha.

Ready for another look.

elle commented:
:shipit:
